plugins/runc/internal/namespace/restore_adapters.go

```diff
@@ -107,19 +107,21 @@ func InheritExternalNamespacesForRestore(nsTypes ...configs.NamespaceType) types
 				if err != nil {
 					return nil, status.Errorf(codes.Internal, "external namespace file %s does not exist: %v", nsPath, err)
 				}
+				defer nsFd.Close()
```

Comment:
Should we close this file before next? Does keeping it open during next affect anything?

Reply:
It needs to be open, so that when CRIU is spawned later, it's also has this file opened. Just tried closing it before, and our tests fail.